cron/format/json_raw_results.go

```diff
@@ -306,3 +327,65 @@ func AsRawJSON(r *pkg.ScorecardResult, writer io.Writer) error {
 
 	return nil
 }
+
+// https://github.com/googleapis/google-cloud-go/blob/bigquery/v1.30.0/bigquery/schema.go#L544
+type bigQueryJSONField struct {
+	Description string              `json:"description"`
+	Fields      []bigQueryJSONField `json:"fields,omitempty"`
+	Mode        string              `json:"mode"`
+	Name        string              `json:"name"`
+	Type        string              `json:"type"`
+}
+
+func generateSchema(schema bigquery.Schema) []bigQueryJSONField {
+	var bqs []bigQueryJSONField
+	for _, fs := range schema {
+		bq := bigQueryJSONField{
+			Description: fs.Description,
+			Name:        fs.Name,
+			Type:        string(fs.Type),
+			Fields:      generateSchema(fs.Schema),
+		}
+		// https://github.com/googleapis/google-cloud-go/blob/bigquery/v1.30.0/bigquery/schema.go#L125
+
+		switch {
+		// Make all fields optional to give us flexibility:
+		// discard `fs.Required`.
+		case fs.Repeated:
+			bq.Mode = "REPEATED"
+		default:
+			bq.Mode = "NULLABLE"
+		}
+
+		bqs = append(bqs, bq)
+	}
+
+	return bqs
+}
+
+// GenerateBqSchema generates the BQ schema in JSON format.
+// Can be used to generate the BQ table:
+// `bq mk --table    --time_partitioning_type DAY \
+// --require_partition_filter=TRUE \
+// --time_partitioning_field date \
+// openssf:scorecardcron.scorecard-rawdata-releasetest \
+// cron/format/bq.raw.schema`.
+func GenerateBqSchema(r *pkg.ScorecardResult, writer io.Writer) error {
```

Comment:
Where does this fn get called from?

Reply:
It's not called. I'm going to revisit this today to see if it should become a utility. Where do you think this should live?

Reply:
PTAL. I reverted most of the updates because BQ does not support pointers to bool, int, etc; but our JSON structure uses that as `nil` to convey "we're not able to tell the value of this field". If I changed our structure to use no pointers, the JSON marshaling will still produce these values. For example, even with `omitempty` on a `bool`, it will be omitted if the value is `false`, which is not what we want.

Conclusion: IFAIK, we cannot share the structure between BQ and JSON. So what I did is provide some utility functions to help us generate these structures in the future.

@azeemshaikh38 PTAL. If the approach sounds reasonable (I just want to have the schema generating code somewhere), I'll merge the PR.